r/R/type.R

```diff
@@ -360,6 +363,12 @@ decimal <- function(precision, scale) {
   } else {
     stop('"precision" must be an integer', call. = FALSE)
   }
+  if (precision > 38) {
+    stop('"precision" must be lower than or equal to 38', call. = FALSE)
+  }
+  if (precision < 1) {
+    stop('"precision" must be greater than 0', call. = FALSE)
+  }
```

Comment:
I know we use `assert_that()` in a lot of the rest of the code - any reason to not use that here?  Asking only out of curiosity as I think it looks better this way anyway.  Feel free to ignore this picky question!

Reply:
The function wasn't using `assert_that()` for the other checks and I went with that.

Reply:
Cool!

Reply:
Happy to go with `assert_that()` if that's the direction, but I think I should switch all arg check to use it. Maybe we can make a follow-up Jira ticket.

Reply:
Nah, it's trivial, I was only asking in case there was a specific thing I didn't know, ultimately it achieves the same thing so don't worry!